### PR TITLE
connected allergens to related models and updated displays

### PR DESCRIPTION
**Pull request by dovebair**
Connected allergies/allergens/severity/people. Currently, it isn't displaying the names of the allergies on the people model, it just says "allergy". Is this because allergies are directly associated with people? Was wondering if it would be better to have an allergy exist independent of people and then they can be assigned as having it in the people model, not the allergy model. Please take a look and let me know what you think.

### CODE REVIEW

File: app/models/allergen.rb [hunk 1]
```diff
@@ -1,2 +1,4 @@
 class Allergen < ApplicationRecord
+  has_many :allergy
```
Comment (jaggederest) [inline app/models/allergen.rb]: `:allergies`, not `:allergy`

File: app/models/allergen.rb [hunk 2]
```diff
@@ -1,2 +1,4 @@
...
+  has_many :people
```
Comment (jaggederest) [inline app/models/allergen.rb]: this is `through: :allergies`

File: app/views/allergies/index.html.erb [hunk 3]
```diff
@@ -15,9 +15,9 @@
   <tbody>
     <% @allergies.each do |allergy| %>
       <tr>
-        <td><%= allergy.allergen %></td>
-        <td><%= allergy.person %></td>
-        <td><%= allergy.severity %></td>
+        <td><%= allergy.allergen.name %></td>
```
Comment (jaggederest) [inline app/views/allergies/index.html.erb]: Optional: We can delegate these, as is done in the quantity model: https://github.com/dovebair/bytr/blob/master/app/models/quantity.rb#L7

Then you can replace `allergy.allergen.name` with `allergy.allergen_name`

File: app/views/people/_form.html.erb [hunk 4]
```diff
@@ -5,6 +5,7 @@
     <%= f.input :name %>
     <%= f.input :birthday %>
     <%= f.input :sex %>
+    <%= f.input :allergy_ids %>
```
Comment (jaggederest) [inline app/views/people/_form.html.erb]: This won't work. You'll need to create allergies on the allergy page, after the person has been created. `localhost:3000/allergies/new` for example

File: app/views/people/index.html.erb [hunk 5]
```diff
@@ -18,6 +19,7 @@
         <td><%= person.name %></td>
         <td><%= person.birthday %></td>
         <td><%= person.sex %></td>
+         <td><%= person.allergies.name %></td>
```
Comment (jaggederest) [inline app/views/people/index.html.erb]: `allergies` is an array. So you'll need to put something on the `Person` model like this:

    def allergy_names
      allergies.pluck(:name).join(',')
    end

And then replace this with `person.allergy_names`

File: db/schema.rb [hunk 6]
```diff
@@ -32,6 +32,14 @@
     t.index ["severity_id"], name: "index_allergies_on_severity_id"
   end
 
+  create_table "consumptions", force: :cascade do |t|
```
Comment (jaggederest) [inline db/schema.rb]: We don't want consumptions any more. Delete this section 🔫

**Comment by dovebair** [comment]
closed in favor of #21

**Pull request closed without merging**